py/mattak/backends/uproot/dataset.py

```diff
@@ -506,19 +510,19 @@ def calibrate(self, waveform_array : numpy.ndarray, starting_window : Union[floa
             calibrated waveform in volt
         """
 
-        channels = list(range(self.num_channels))
+        channels = channels or list(range(self.num_channels))
```

Comment:
If a number is allowed to select a channel, then 0 won't work because 0 is falsey. If you intend that a single number may be passed (instead of e.g. [0] or (0,)), it may be worth automatically converting 0 to (0,) before here (otherwise, the truthy check is good because it will detect things like empty lists).

Reply:
yes that was my idea